qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -169,3 +184,62 @@ def get(self):
         users, links = yield Task(self._get_shared_for_study, study)
 
         self.write(dumps({'users': users, 'links': links}))
+
+
+class SearchStudiesAJAX(BaseHandler):
+    def _get_standard(self, user, callback):
+        callback(_build_study_info("standard", user))
+
+    def _get_shared(self, user, callback):
+        """builds list of tuples for studies that are shared with user"""
+        callback(_build_study_info("shared", user))
+
+    @authenticated
+    def get(self, ignore):
+        search_type = self.get_argument('type')
+        user = self.get_argument('user')
+        query = self.get_argument('query')
+        echo = int(self.get_argument('sEcho'))
+
+        res = None
+        if query != "":
+            # Search for samples matching the query
+            search = QiitaStudySearch()
+            try:
+                res, meta = search(query, User(user))
+            except ParseException:
+                self.clear()
+                self.set_status(400)
+                self.write('Malformed search query. Please read "search help" '
+                           'and try again.')
+                return
+            except QiitaDBIncompatibleDatatypeError as e:
```

Comment:
Under what types of situations would this arise? I didn't see a test case on it, so just curious